### PR TITLE
Add initial port of unified scripts: utils and transferor (DO NOT MERGE)

### PR DESCRIPTION
**Pull request by vkuznet**
This implementation ports WmAgentScripts/Unified
- port utils.py and trasferor.py scripts
- code refactoring and clean-up
- new components:
  - Common.py # set of common utilities for Unified
  - SiteInfo.py # site information module
  - RequestInfo.py # request information module
  - Transferor.py # transferor.py script
  - Stageout.py # initial placeholder for stageout module
  - Injector.py # initial placeholder for Injector module

I run local pylint rules and adjusted all issues with original code.

### CODE REVIEW

**Comment by vkuznet** [comment]
@ticoann , @amaltaro , @ericvaandering Please review initial commit for transferor module. Please fire up jenkins tests too since I need its input for pylint/pycode/pep/etc. tests. Here is a link to its output: http://bit.ly/2wvqAJR

I heavily rely on pycurl multi fetch url and sso I committed earlier. I was able to get all transferor info (dbs, phedex, sites, monitoring, detox, etc.) under 400 sec. Now code is ready to start serious refactoring towards better throughput. So far, I didn't yet place any phedex requests though. But in aforementioned link you'll see times breakdown for various components. So far the major time contributor is SiteInfo cache and getting site list for all workflows.

**Comment by ericvaandering** [comment]
Not sure what's up with Jenkins. The whole service seems to be down at the moment. But Valentin, you must be able to write a test that starts up the micro service on a port and tests it, right?

**Comment by vkuznet** [comment]
Eric,
I plan to add plenty of tests, the Unified codebase is huge and hard to follow.
My initial attempt is code re-factoring and reaching base functionality.
Then we need to setup testbed with workflows to tests and only then we can
test things properly. Right now I'll need jenkins feedback about code quality.

**Comment by ericvaandering** [comment]
You can get that yourself with pylint with standards/.pylintrc and pep8. But I know, it's easier to wait for Jenkins.

Might be worth to have some simple tests that just test the micro service concept without testing the particular logic inside.

**Comment by ericvaandering** [comment]
test this please

**Comment by ericvaandering** [comment]
Whoah, what happened here? It looks like you rebased incorrectly, @vkuznet

**Comment by vkuznet** [comment]
I messed with tip of the branches and re-applied changes in #8099  Closing this one and will clean-up my tree.
